Documentation: Update Storage docs - Move or Copy a Volume [WD-21424] (#15551)
canonical/lxd: doc/howto/storage_move_volume.md

Reviewer: There is a sentence refering to the move button, where it should refer to the "migrate" button:

"Navigate to the overview page of the selected instance, and click on the Move button in the top right corner."

Author: The sentence in question appears in the context of moving instance root volumes to pools, which we already decided would use Move terminology rather than Migrate (Only custom volumes and moving instance root storage across clusters use Migrate, now)

Reviewer: Probably easier to explain with the full context. There is no "move" button in the instance header. I think we should refer to the "Migrate" button here:

![image](https://github.com/user-attachments/assets/2ed144e7-bd8e-422f-ad55-ab05125be287)

When copying from one storage pool to another, you can either use the same name for both volumes or rename the new volume.

````
````{group-tab} UI
To copy a custom storage volume, navigate to the {guilabel}`Overview` page of the storage volume you wish to copy, and click {guilabel}`Copy`.

> ```{figure} /images/storage/storage_volumes_overview.png

Reviewer 2: Need to either replace or remove this image as it still says Duplicate.

When moving from one storage pool to another, you can either use the same name for both volumes or rename the new volume.

````
````{group-tab} UI

To rename a custom storage volume, navigate to its {guilabel}`Overview` page and select its name in the header to edit it.

```{figure} /images/storage/storage_volumes_rename.png
:width: 60%
:alt: LXD Rename Custom Storage Volume
```

````
`````

## Copy or migrate between cluster members

`````{tabs}
````{group-tab} CLI

For most storage drivers (except for `ceph` and `ceph-fs`), storage volumes exist only on the cluster member for which they were created.

To copy or migrate a custom storage volume from one cluster member to another, add the `--target` and `--destination-target` flags to specify the source cluster member and the target cluster member, respectively.

````
````{group-tab} UI

> You can use the LXD UI to copy storage volumes between cluster members, but not to move them.

Reviewer 2: ```suggestion
You can use the LXD UI to copy storage volumes between cluster members, but not to migrate them.
```

When moving from one storage pool to another, you can either use the same name for both volumes or rename the new volume.

````
````{group-tab} UI

You can use the LXD UI to rename a custom storage volume, but not to move it.

To rename a custom storage volume, navigate to its {guilabel}`Overview` page and select its name in the header to edit it.

```{figure} /images/storage/storage_volumes_rename.png
:width: 60%
:alt: LXD Rename Custom Storage Volume
```

````
`````

## Copy or migrate between cluster members

`````{tabs}
````{group-tab} CLI

For most storage drivers (except for `ceph` and `ceph-fs`), storage volumes exist only on the cluster member for which they were created.

> To copy or move a custom storage volume from one cluster member to another, add the `--target` and `--destination-target` flags to specify the source cluster member and the target cluster member, respectively.

Reviewer 2: ```suggestion
To copy or migrate a custom storage volume from one cluster member to another, add the `--target` and `--destination-target` flags to specify the source cluster member and the target cluster member, respectively.
```

````
````{group-tab} UI

> You can use the LXD UI to copy storage volumes between cluster members, but not to move them.

Reviewer 2: ```suggestion
You can use the LXD UI to copy storage volumes between cluster members, but not to migrate them.
```

> You can use the LXD UI to migrate storage volumes between cluster members, but not to move them.

Reviewer: This seems wrong, I think we should say "can copy volumes between members, but not move them".

Author: Reworded to "You can use the LXD UI to copy storage volumes between cluster members, but not to move them."

To copy a storage volume, navigate to the {guilabel}`Overview` page of the storage volume within a clustered environment, then click {guilabel}`Copy`.

> In the {guilabel}`Copy volume` modal, select an alternative cluster member from the {guilabel}`Cluster member` dropdown.

Reviewer 2: ```suggestion
In the {guilabel}`Copy volume` modal, select the target cluster member from the {guilabel}`Cluster member` dropdown.
```

````
`````

## Copy or move between projects

`````{tabs}
````{group-tab} CLI

Add the `--target-project` to copy or move a custom storage volume to a different project.

````
````{group-tab} UI

To copy a storage volume between projects, navigate to the {guilabel}`Overview` page of the storage volume, then click {guilabel}`Copy`.

> In the {guilabel}`Copy volume` modal, select an alternative project from the {guilabel}`Target project` dropdown.

Reviewer 2: ```suggestion
In the {guilabel}`Copy volume` modal, select the target from the {guilabel}`Target project` dropdown.
```

> In the {guilabel}`Copy volume` modal, select the target from the {guilabel}`Target project` dropdown.

Reviewer: I think this is not yet supported in the UI.

Author: ![image](https://github.com/user-attachments/assets/5a628702-f35c-47ce-b552-25930323455a)
I do believe one can copy custom storage volumes across projects using the modal provided above. Is this an incorrect interpretation?

When moving from one storage pool to another, you can either use the same name for both volumes or rename the new volume.

````
````{group-tab} UI

> 

Reviewer: This seems only partially right. We can move the volume to another pool in the UI.

> 

Author: I have taken out this line.

````
````{group-tab} UI

To copy or move a storage volume between cluster members, open the {guilabel}`Copy volume` modal by navigating to the overview page of the storage volume you wish to copy, and clicking the {guilabel}`Copy` button in the top right.

In the {guilabel}`Copy volume` modal, select an alternative cluster member from the {guilabel}`Cluster member` dropdown.

When you have finished configuring the new storage volume, click {guilabel}`Copy` to finalize the operation.

````
`````

## Copy or move between projects

`````{tabs}
````{group-tab} CLI

Add the `--target-project` to copy or move a custom storage volume to a different project.

````
````{group-tab} UI

To copy or move a storage volume between projects, open the {guilabel}`Copy volume` modal by navigating to the overview page of the storage volume you wish to copy, and clicking the {guilabel}`Copy` button in the top right.

In the {guilabel}`Copy volume` modal, select an alternative project from the {guilabel}`Target project` dropdown.

> 

Reviewer 2: ```suggestion
```

> When you have finished configuring the new storage volume, click {guilabel}`Copy` to finalize the operation.

Reviewer 2: ```suggestion
```
